### PR TITLE
Implement newtype IDs

### LINKED ISSUE #84: Newtypes for IDs
Seems to me it would be nice to have newtypes for the different IDs (orgs, users, check runs, workflow runs, etc). That would be an API break though. How do you feel about that?

### PR DESCRIPTION
**Pull request by rocallahan**
Resolves #84. Bumps semver version because this is a compatibility break.

### CODE REVIEW

**Comment by rocallahan** [comment]
This PR defines ID newtypes and uses them in the models. It doesn't update APIs to use them. I want to do that, but I have a question:

Some APIs support generic `Into<u64>` ID parameters, e.g.
```
    pub async fn get(&self, id: impl Into<u64>) -> crate::Result<Notification> {
```
I'm not sure how that was intended to be used. If we translate this to
```
    pub async fn get(&self, id: impl Into<NotificationId>) -> crate::Result<Notification> {
```
that's not very useful because currently the only type that converts into `NotificationId` is `NotificationId` itself. We could add a conversion from raw `u64` to `NotificationId` but I suggest we don't because that undermines the value of newtypes here. So is it OK to just replace these generic parameters with the newtype?

**Comment by XAMPPRocky** [comment]
Yeah, I would replace the `impl Into` with the newtype and add a `From<u64>` impl, that way you have to explicitly opt-in to converting the raw id into the newtype.

**Comment by rocallahan** [comment]
BTW Github is waiting for you to approve running tests on my PR.

**Comment by XAMPPRocky** [comment]
> BTW Github is waiting for you to approve running tests on my PR.

Huh, never seen that before. I've approved the run.

**Comment by rocallahan** [comment]
See https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/

File: src/models.rs
Comment (XAMPPRocky) [inline src/models.rs:49]:
```suggestion
    ($($name:ident),+) => {$(
        #[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
        pub struct $name(pub BaseIdType);
        impl fmt::Display for $name {
            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
                self.0.fmt(f)
            }
        }
        impl Deref for $name {
            type Target = BaseIdType;
            fn deref(&self) -> &Self::Target {
                &self.0
            }
        }
        impl DerefMut for $name {
            fn deref_mut(&mut self) -> &mut Self::Target {
                &mut self.0
            }
        }
        impl $name {
            pub fn into_inner(self) -> BaseIdType {
                self.0
            }
        }
    )+};
```

Can you change this macro to accept multiple idents and generate definitions for each one. This removes needing to write `id_type` for each one.

File: src/models.rs
Comment (XAMPPRocky) [inline src/models.rs:52]: typo
```suggestion
id_type!(ActorId); // A Bot, EnterpriseUserAccount, Mannequin, Organization or User
```

File: src/models.rs
Comment (XAMPPRocky) [inline src/models.rs:81]: Ditto with this macro. I would probably have a syntax like `OrgId -> ActorId, UserId -> ActorId`. It'd look like this in the macro `($($from:ident -> $to:ident),+)`

File: src/models.rs
Comment (XAMPPRocky) [inline src/models.rs:48]: Could you also add an `AsRef<BaseIdType>` implementation?

**Review by XAMPPRocky — CHANGES_REQUESTED** [review]
Overall looks good to me, I've left a couple comments about small improvements.

**Comment by rocallahan** [comment]
Oops didn't see your comments, I'll address them now

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**
